test/common/http/http2/codec_impl_test.cc

```diff
@@ -360,6 +360,30 @@ TEST_P(Http2CodecImplTest, ShutdownNotice) {
   response_encoder_->encodeHeaders(response_headers, true);
 }
 
+TEST_P(Http2CodecImplTest, ProtocolErrorForTest) {
```

Comment:
I think you cover this now in the integration test? Do we need this? Then the hack with the dynamic cast is only in one place.

Reply:
I needed it because the code coverage for source/common/http/http2 dropped too low without it